components/tor/tor_file_watcher_unittest.cc

```diff
@@ -0,0 +1,191 @@
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+  tor_file_watcher.reset(
+      new TorFileWatcher(test_data_dir().AppendASCII("auth_cookies_too_long")));
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlCookie(cookie, time));
+  EXPECT_EQ(cookie.size(), 0u);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+  constexpr unsigned char expected_auth_cookie[] = {
+      0x6c, 0x6e, 0x9d, 0x24, 0x78, 0xe6, 0x6d, 0x69, 0xd3, 0x2d, 0xc9,
+      0x90, 0x9a, 0x3c, 0x39, 0x54, 0x2b, 0x37, 0xff, 0x30, 0xda, 0x5a,
+      0x90, 0x94, 0x44, 0xa4, 0x3d, 0x30, 0xd5, 0xa9, 0x19, 0xef};
+  unsigned int expected_auth_cookie_len = 32;
+
+  tor_file_watcher.reset(
+      new TorFileWatcher(test_data_dir().AppendASCII("normal_auth_cookies")));
+  tor_file_watcher->polling_ = true;
+  EXPECT_TRUE(tor_file_watcher->EatControlCookie(cookie, time));
+  EXPECT_EQ(std::memcmp(cookie.data(), expected_auth_cookie,
+                        expected_auth_cookie_len),
+            0);
+  EXPECT_EQ(cookie.size(), expected_auth_cookie_len);
+  EXPECT_NE(time.ToJsTime(), 0u);
+}
+
+TEST_F(TorFileWatcherTest, EatControlPort) {
+  int port = -1;
+  base::Time time;
+
+  std::unique_ptr<TorFileWatcher> tor_file_watcher =
+      std::make_unique<TorFileWatcher>(
+          test_data_dir().AppendASCII("not_valid"));
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlPort(port, time));
+  EXPECT_EQ(port, -1);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+  // controlport is a folder
+  tor_file_watcher.reset(new TorFileWatcher(test_data_dir()));
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlPort(port, time));
+  EXPECT_EQ(port, -1);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+  tor_file_watcher.reset(
+      new TorFileWatcher(test_data_dir().AppendASCII("empty_controlport")));
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlPort(port, time));
+  EXPECT_EQ(port, -1);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+#if defined(OS_WIN)
+  tor_file_watcher.reset(new TorFileWatcher(
+      test_data_dir().AppendASCII("invalid_controlport_win")));
+#else
+  tor_file_watcher.reset(
+      new TorFileWatcher(test_data_dir().AppendASCII("invalid_controlport")));
+#endif
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlPort(port, time));
+  EXPECT_EQ(port, -1);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+#if defined(OS_WIN)
+  tor_file_watcher.reset(new TorFileWatcher(
+      test_data_dir().AppendASCII("valid_controlport_not_localhost_win")));
+#else
+  tor_file_watcher.reset(new TorFileWatcher(
+      test_data_dir().AppendASCII("valid_controlport_not_localhost")));
+#endif
+  tor_file_watcher->polling_ = true;
+  EXPECT_FALSE(tor_file_watcher->EatControlPort(port, time));
+  EXPECT_EQ(port, -1);
+  EXPECT_EQ(time.ToJsTime(), 0u);
+
+#if defined(OS_WIN)
+  tor_file_watcher.reset(new TorFileWatcher(
+      test_data_dir().AppendASCII("controlport_too_long_win")));
```

Comment:
for easier readability can you make a const for these with the os flags?

Reply:
fixed in 730f9927c864fb2d71e5f9089fc107f7b351fc8b